echidna/core/spectra.py

```diff
@@ -229,3 +229,40 @@ def add(self, spectrum):
         self._data += spectrum._data
         self._raw_events += spectrum._raw_events
         self._num_decays += spectrum._num_decays
+
+    def rebin(self, new_bins):
+        """ Rebin spectra data into a smaller spectra of the same rank whose
+        dimensions are factors of the original dimensions.
+
+        Args:
+          new_bins (tuple): New bin sizes for spectra. Should be in order of
+            energy, radial, time.
+
+        Raises:
+          ValueError: Shape mismatch. Number of dimenesions are different.
+          ValueError: Old bins/ New bins must be integer
+        """
+        if self._data.ndim != len(new_bins):
+            raise ValueError("Shape mismatch: %s->%s" % (self._data.shape,
+                                                         new_bins))
+        for i in range(len(new_bins)):
+            if self._data.shape[i] % new_bins[i] != 0:
+                raise ValueError("Old bins/New bins must be integer old: %s"
+                                 " new: %s" % (self._data.shape, new_bins))
+        compression_pairs = [(d, c//d) for d, c in zip(new_bins,
+                                                       self._data.shape)]
+        flattened = [l for p in compression_pairs for l in p]
+        self._data = self._data.reshape(flattened)
+        for i in range(len(new_bins)):
+            self._data = self._data.sum(-1*(i+1))
+        self._energy_bins = new_bins[0]
+        self._radial_bins = new_bins[1]
+        self._time_bins = new_bins[2]
+        self.calc_widths()
+
+    def calc_widths(self):
```

Comment:
Could make this an internal method `_calc_widths` and call it from `__init__` as well.